features/268-next-cf/api.md

```diff
@@ -664,3 +664,97 @@ function. Controller will check the mapping and create fission function with
 k8s secrets. Fission support reading secrets value from files. We will update
 the fission environment runtime to convert those values to environment variable,
 so skygear user can access those secret form env.
+
+## Batch deploy items in skygear.yaml
+
+`skycli app deploy` will deploy all deployment items in `skygear.yaml`.
+`skycli app deploy [ITEM_NAME]` will only create or update item with name
+`[ITEM_NAME]`.
+
+When user use `skycli app deploy`, they need to have all items in the
+`skygear.yaml`. The deployment will not keep the previously deployed item, if
+they don't exist in `skygear.yaml`. e.g. User had deployed function 1 to 4. If
+user run deploy with a config file that only has function 1 to 2. The new
+version app will only have function 1 to 2. User has responsibility to keep the
+full version of `skygear.yaml` and codes.
+
+User can create or update particular item by `skycli app deploy [ITEM_NAME]`.
+e.g. User had deployed function 1 to 4. If user use
+`skycli app deploy function5` to deploy function 5, the new app version will
+have function 1 to 5 after deployment.
+
+### Implementation Details
+
+#### Models
+
+**`app_version`**
+
+Every deployment (one or multiple items) will create a new app_version.
+
+- `id`
+- `created_at`: deployment time
+- `app_id`: skygear app id
+- `version`: deployment version, uuid or hash
+- `tag`: deployment tag, the live version will have tag with `latest`
```

Comment:
We need a status column here because the below flow references it. It would be great if we can write down the possible values.

We may need think about the relationship between app_version.status and cloud_code.status.

Reply:
Yes, i missed it, let me add it back!